### PR TITLE
entc/gen: exposing filter/where api on mutations

### PR DESCRIPTION
**Pull request by a8m**
cc @zeevmoney

### CODE REVIEW

File: examples/version/ent/schema/user.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	return []ent.Hook{
+		// Apply the `OptimisticLock` hook only on `UpdateOne` operations,
+		// and block all `Update` (update-many) operations as we don't have
+		// access to the nodes that are affected by these mutation.
+		hook.On(OptimisticLock(), ent.OpUpdateOne),
+		hook.Reject(ent.OpUpdate),
+	}
+}
+
+func OptimisticLock() ent.Hook {
```
Comment (a8m) [inline examples/version/ent/schema/user.go:71]: cc @zeevmoney, @rotemtam

File: examples/version/example_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	// The operation fails because the user was updated by another process (usrCopy).
+	_, err = usr.Update().SetStatus(user.StatusOffline).Save(ctx)
+	fmt.Println(err)
+
+	// Output:
+	// 1 online
+	// 1 offline
+	// user record was changed by another process: ent: user not found
+}
```
Comment (a8m) [inline examples/version/example_test.go:46]: cc @zeevmoney, @rotemtam

File: examples/version/ent/schema/user.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	return func(next ent.Mutator) ent.Mutator {
+		// A hook that validates the "version" field is incremented by 1 on each update.
```
Comment (rotemtam) [inline examples/version/ent/schema/user.go:73]: I think the comment here is wrong?

File: examples/version/example_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	fmt.Println(usrCopy.ID, usrCopy.Status)
+
+	// The operation fails because the user was updated by another process (usrCopy).
+	_, err = usr.Update().SetStatus(user.StatusOffline).Save(ctx)
+	fmt.Println(err)
+
+	// Output:
+	// 1 online
+	// 1 offline
+	// user record was changed by another process: ent: user not found
```
Comment (rotemtam) [inline examples/version/example_test.go:45]: "ent: user not found" is a bit unclear/misleading, what does it mean?

**Pull request merged**